ipld/blockstore/src/cbor.rs

```diff
@@ -1,7 +1,7 @@
 use cid::{multihash, Cid};
+use fvm_ipld_encoding::{de, from_slice, ser, to_vec, DAG_CBOR};
 
 use super::{Block, Blockstore};
-use crate::encoding::{de, from_slice, ser, to_vec, DAG_CBOR};
```

Comment:
Hm. I'd still rather not have the blockstore abstractions depend on serde... but it's not terrible because we use it everywhere anyways. We can always put it behind a feature flag later.

Reply:
Ah, no, this is a problem. We need to return the correct `Error` type lest we break error downcasting in the actors.

Reply:
For our current magical error conversion in actors, yes. We should probably have tests for this specific case.

Reply:
@Stebalien asfiu we should be able to use `serde_ipld_dagcbor::error::Error` in the downcast chain instead or in addition at least to `encoding::Error`, for downcasting, and things "should just work". thoughts?

Reply:
That technically works, but we _really_ wanted to avoid that as it makes it harder to swap out the underlying IPLD library. If we need to do decoding in the blockstore library, we'd rather use our encoding shim.

Hence the suggestion to move the cbor blockstore into encoding itself (it's the "encoding" blockstore) but I'd also be happy to re-add the dependency from blockstore to encoding if that seems cleaner to you.

Reply:
Moved the cborstore into the encoding crate, it makes more sense this way, I agree after thinking about it some more